### ISSUE TITLE
Ray installation fails on CentOS.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#813]
I'm testing this on the default Amazon Linux AMI.

First I had to do `pip install cmake` to get a sufficiently recent version of cmake. Then I ran into this issue.

```
-- Checking for module 'arrow'
--   No package 'arrow' found
CMake Error at cmake_modules/FindArrow.cmake:119 (message):
  Could not find the Arrow library.  Looked for headers in , and for libs in
Call Stack (most recent call first):
  CMakeLists.txt:157 (find_package)
```

The issue seems to be that we hardcode the path `PKG_CONFIG_PATH=$ARROW_HOME/lib/pkgconfig` in `src/thirdparty/build_thirdparty.sh`, but on this machine it was generated as `$ARROW_HOME/lib64/pkgconfig`.

As a temporary solution, doing `cp -r $ARROW_HOME/lib64 $ARROW_HOME/lib` solved the problem.

**Comment by robertnishihara** [comment]
This should be more or less fixed by #840, but getting Ray to work on a fresh CentOS machine still isn't straightforward. You probably have to upgrade cmake and maybe do a few other things.

**Comment by robertnishihara** [comment]
Closing for now since it should work (e.g., we compile our wheels on the manylinux image). However, there are probably a number of annoying things that have to be done to build Ray on a fresh centos image (like upgrading cmake), but I'm not sure what they are so those should be tracked in a different issue (perhaps the issue should be to document what steps need to be taken).
